application/src/test/java/uk/nhs/hee/tis/revalidation/changelog/SetDraftRecommendationsActualSubmissionDatesToNullChangeLogTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.nhs.hee.tis.revalidation.changelog;
+
+import java.time.LocalDate;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.nhs.hee.tis.revalidation.entity.Recommendation;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.READY_TO_REVIEW;
+
+@ExtendWith(MockitoExtension.class)
+class SetDraftRecommendationsActualSubmissionDatesToNullChangeLogTest {
+
+  @InjectMocks
+  SetDraftRecommendationsActualSubmissionDatesToNullChangeLog changeLog;
+
+  @Mock
+  RecommendationRepository recommendationRepository;
+
+  private static final String RECOMMENDATION_ID = "123";
+
+  private Recommendation draftWithDate = buildDraftWithActualSubmissionDate();
+  private Recommendation draftWithoutDate = buildDraftWithoutActualSubmissionDate();
+
+  @Test
+  void shouldNotUpdateDraftIfActualSubmissionDateNull() {
+    when(recommendationRepository.findAllByRecommendationStatus(READY_TO_REVIEW)).thenReturn(
+        List.of(draftWithoutDate)
+    );
+    changeLog.setActualSubmissionDatesToNull(recommendationRepository);
+    verify(recommendationRepository, never()).save(draftWithoutDate);
+  }
+
+  @Test
+  void shouldUpdateDraftIfActualSubmissionDateNotNull() {
+    when(recommendationRepository.findAllByRecommendationStatus(READY_TO_REVIEW)).thenReturn(
+        List.of(draftWithDate)
+    );
+    changeLog.setActualSubmissionDatesToNull(recommendationRepository);
+    verify(recommendationRepository).save(draftWithoutDate);
```

Comment:
Is `draftWithoutDate` rather than `draftWithDate` some #copy:spaghetti:?  If so, worth looking at why the test passes!

It might be that you want to use a fakers/random for the recommendation id field.

Reply:
Here I think we need to verify the post parameter doesn't have the actualSubmissionDate, so `draftWithoutDate` is the expected one. But, either `draftWithoutDate` or `draftWithDate` can pass the tests!!

Because Mockito uses `Recommendation.equals` to check if the object matches. And the `equals()` method is generated by Lombok `@Data`, here is the doc: https://projectlombok.org/features/Data (Scrolling down to `equals()`, we can see it checks the reference first, if references don't match, then check eatch fields).